src/buildercore/terraform.py

```diff
@@ -1229,6 +1229,46 @@ def _render_eks_master_security_group(context, template):
         'tags': security_group_tags,
     })
 
+def _render_eks_addon(context, template, name, addon_name, version, configuration_values=False, resolve_conflicts='OVERWRITE', service_account_role_arn=False):
+    if (version == 'latest'):
+        template.populate_data(
+            'aws_eks_addon_version',
+            'eks_addon_%s' % name,
+            block={
+                'addon_name': addon_name,
+                'kubernetes_version': '${data.aws_eks_cluster.main.version}',
+                'most_recent': True,
+            }
+        )
+
+    resource_block = {
+        'cluster_name': '${data.aws_eks_cluster.main.id}',
+        'addon_name': addon_name,
+        'addon_version': version if version != 'latest' else '${data.aws_eks_addon_version.eks_addon_%s.version}' % name,
+        'tags': aws.generic_tags(context),
+        'resolve_conflicts': resolve_conflicts,
+    }
+
+    if (configuration_values != False):
+        resource_block['configuration_values'] = configuration_values
+
+    if (service_account_role_arn != False):
+        resource_block['service_account_role_arn'] = service_account_role_arn
+
+    template.populate_resource(
+        'aws_eks_addon',
+        'eks_addon_%s' % name,
+        block=resource_block,
+    )
+
+def _render_eks_addons(context, template):
+    if (lookup(context, 'eks.addons.kube-proxy')):
+        addon_version = lookup(context, 'eks.addons.kube-proxy.version', 'latest')
```

Comment:
`terraform.py` should rely on the context building that happens in `cfngen.py` to supply these defaults. If the values it needs are not present, it should fail rather than try to fallback to other values.

This is the *intent* anyway, a separation of concerns, but it's not always followed and it certainly wasn't followed in the beginning when the logic for `builder` was still condensing.

Reply:
You can't fix (or review or learn from) non-existent code. Ploughing into it is the best way sometimes, don't worry about it.

> I think the interaction between config and the generated output in terraform will be a little more complicated

It's not complicated *right now* so we can save that engineering effort for another PR.

> I'll maybe fire a PR your way once I've thought about it a bit more today.

More than happy to review changes to `builder`, it helps keep my mental model up to date.